services/api-server/src/simcore_service_api_server/api/errors/custom_errors.py

```diff
@@ -0,0 +1,31 @@
+from urllib.request import Request
+
+from fastapi import status
+from starlette.responses import JSONResponse
+
+
+class CustomBaseError(Exception):
+    pass
+
+
+class InsufficientCredits(CustomBaseError):
+    pass
+
+
+class MissingWallet(CustomBaseError):
+    pass
+
+
+async def custom_error_handler(_: Request, exc: CustomBaseError):
+    if isinstance(exc, InsufficientCredits):
+        return JSONResponse(
+            status_code=status.HTTP_402_PAYMENT_REQUIRED, content=f"{exc}"
+        )
+    if isinstance(exc, MissingWallet):
+        return JSONResponse(
+            status_code=status.HTTP_424_FAILED_DEPENDENCY, content=f"{exc}"
+        )
+    return JSONResponse(
+        status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+        content="Encountered unspecified error",
```

Comment:
I see your point. But the point is that this case should never happen. This only happens if we/I don't properly handle all the different errors on the server. So I wouldn't know what else to write here. Of course I could write something like "Please submit an issue to explain that you hit this issue on Github". Would that be better?

Reply:
No, because AFAIK the error is for the service developers and not te client. Then I would add an `_logger.exception` and return a `OEC` to the client as of "Unexpected error on the server [OEC: ****]`. As an example of how to create this you can find https://github.com/itisfoundation/osparc-simcore/blob/6c97c5e8aa0e246c695133284b3c8712da1fa28e/packages/service-library/tests/test_error_codes.py#L33